terraform/lib/dependabot/terraform/file_parser.rb

```diff
@@ -49,29 +48,51 @@ def parse_terraform_file(file)
       end
 
       def parse_terragrunt_file(file)
-        modules = parsed_file(file).fetch("terraform", []).first || {}
-        modules.each do |details|
-          next unless details.is_a?(Hash)
-          next unless details.key?("terraform")
-            @dependency_set << build_terragrunt_dependency(file, details)
+        modules = parsed_file(file).fetch("terraform", []).
+                  map { |k, v| [k, v] }.to_h
+        modules.each do |name, details|
+          next unless name == "source"
+
+          @dependency_set << build_terraform_dependency(
+            file, "", [Hash[name, details]]
+          )
         end
       end
 
       def parse_terragrunt_legacy_file(file)
-        modules = parsed_file(file).fetch("terragrunt", []).first || {}
-        modules = modules.fetch("terraform", [])
-        modules.each do |details|
-          next unless details["source"]
+        module_list = parsed_file(file).fetch("terragrunt", [])
```

Comment:
It seems like there should be some simple optimizations possible by noting that:

* In the legacy terraform.tfvars terragrunt configuration, there can be a single `terragrunt` block, and with the `terragrunt` block there can be a single `terraform` block. So while these are lists, they are lists of a single item.

* In the new terragrunt.hcl, there is *no* `terragrunt` block, but the `terraform` block is the same as the legacy approach. There can be only one.

I don't know if it is worth simplifying/optimizing the code with that in mind, considering our limited experience with Ruby. You tell me if you'd like to give it a try.